Feature: Delete Worker Pool

khoaguin opened the pull request:
Issue: https://github.com/OpenMined/Heartbeat/issues/916

khoaguin committed:
[syft/container_workload] draft version for `SyftWorkerPoolService.delete`

khoaguin committed:
Merge branch 'dev' into delete-worker-pool

khoaguin committed:
[syft/container_workload] deleting a worker pool
- make `job_id` as a searchable attribute in `QueueItem`
- add the `get_by_job_id` method to `QueueStash`
- add the `get_by_job_id` and `update_queue_status` methods to `QueueService`
- add the `delete` method to `SyftWorkerPoolService`
- for `worker_pool_service_test`:
	- add deleting worker pool in existing tests
	- add a nested syft function and deleting working pool test

khoaguin committed:
Merge branch 'dev' into delete-worker-pool

yashgorana commented:
@shubham3121 we'll need to have this supported on kubernetes

khoaguin committed:
[syft/container_workload] delete worker pools in some tests

khoaguin committed:
Merge branch 'dev' into delete-worker-pool

khoaguin committed:
[syft/container_workload] - improve unit tests
- pass `syft_node_location` and `syft_client_verify_key` to
workers in `_create_workers_in_pool`

khoaguin committed:
[syft/container_workload] delete worker pool service works
if there are no running jobs on the workers

shubham3121 commented:
Yes @yashgorana , we will want to do that same in kubernetes as well, but once the kubernetes PR is merged.

khoaguin committed:
Merge branch 'dev' into delete-worker-pool

khoaguin committed:
Merge branch 'dev' into delete-worker-pool

khoaguin committed:
[syft/container_workload] - refactor worker pool delete service
- improve print message when setting up default worker pool

khoaguin committed:
[syft/container_workload] update job id for a worker when it's consuming the job

khoaguin committed:
[syft/container_workload] cleaning up unused services
- return `WorkerPoolDeletionStatus` for the worker pool delete service

khoaguin committed:
[tests/container_workload] test delete worker pool in integration stack tests

khoaguin committed:
[syft/container_workload] get the pool's workers by calling
`LinkedObject.resolve_with_context` instead of `.resolve`

Co-authored-by: Shubham Gupta <shubhamgupta3121@gmail.com>

khoaguin committed:
Merge branch 'dev' into delete-worker-pool

khoaguin committed:
[tests/container_workload] skip checking the number of
worker pools after deletion in integration tests since
we use the same container-stack domain for these tests

khoaguin committed:
[tests/container_workload] (cont) skip checking the number of
worker pools after deletion in integration tests since
we use the same container-stack domain for these tests

khoaguin committed:
Merge branch 'dev' into delete-worker-pool

khoaguin committed:
Merge branch 'dev' into delete-worker-pool

khoaguin committed:
Merge branch 'dev' into delete-worker-pool

khoaguin commented:
Closed as it's outdated and we need to also make it work with Kubernetes, so mostly will be an entirely rework
